File: AUGenExport-v03/OSX/AUGenExportFramework/AUGenExportView.m
/*
	Copyright (C) 2016 Apple Inc. All Rights Reserved.
	See LICENSE.txt for this sample’s licensing information
	
	Modified by Sam Tarakajian --- starakajian@gmail.com
	
	Abstract:
	View for the AUGenExport audio unit.
*/

#import "AUGenExportView.h"

@interface AUGenExportView () {
    CALayer         *containerLayer;
}
@end

@implementation AUGenExportView

-(void)dealloc
{
    NSLog(@"Dealloc called on AUGenExportView\n");
}

-(void)awakeFromNib {
    
    containerLayer = [CALayer layer];
    containerLayer.name = @"container";
    containerLayer.anchorPoint = CGPointZero;
    containerLayer.frame = CGRectMake(0, 0, self.layer.bounds.size.width, self.layer.bounds.size.height);
    containerLayer.bounds = containerLayer.frame;
	containerLayer.backgroundColor = [NSColor greenColor].CGColor;
    [self.layer addSublayer: containerLayer];
}

@synthesize delegate;

@end

File: AUGenExport-v03/iOS/AUGenExportFramework/AUGenExportView.swift
/*
	Copyright (C) 2016 Apple Inc. All Rights Reserved.
	See LICENSE.txt for this sample’s licensing information
	
	Abstract:
	View for the AUGenExport audio unit. It doesn't do anything.
*/

import UIKit

class AUGenExportView: UIView {
    
    var containerLayer = CALayer()
    
    override func awakeFromNib() {
        let scale = UIScreen.mainScreen().scale
        
        containerLayer.name = "container"
        containerLayer.anchorPoint = CGPoint.zero
        containerLayer.frame = CGRect(origin: CGPoint.zero, size: layer.bounds.size)
        containerLayer.bounds = containerLayer.frame
        containerLayer.contentsScale = scale
		containerLayer.backgroundColor = UIColor.blueColor().CGColor;
        layer.addSublayer(containerLayer)

        layer.contentsScale = scale
    }
	
	override func layoutSublayersOfLayer(layer: CALayer) {        
        if layer === self.layer {
            CATransaction.begin()
            CATransaction.setDisableActions(true)
            
            containerLayer.bounds = layer.bounds
			
            CATransaction.commit()
        }
    }
    
    override func touchesBegan(touches: Set<UITouch>, withEvent event: UIEvent?) {
		NSLog("Touch Down");
    }
}